### PR TITLE
Resolve #288 Mobile Design for Weigh In

### LINKED ISSUE #288: Mobile Design for Weigh In
**Describe the feature. What should a user be able to do?**
Need an XD design of what Weigh In will look like in mobile and the "medium" transition stage.

**Alternatives**
Have discussed a one or two column layout.

### PR DESCRIPTION
**Pull request by smithWEBtek**
Resolves #288 Mobile Design for Weigh In.

# Why is this change necessary?
This change was necessary to conform to the design required.

# How does it address the issue?
* Javascript is added for populating flex grid of weigh-in responses.
* JQuery is added for truncating response text (line clamping / ellipsis effect).
* Serializer and JSON response code added to Refinery Stories controller to serve up JSON data for weigh-in stories
* Javascript is used for storing story JSON data in browser local storage for re-populating response text between screen sizes. 
* CSS is added for styling at various screen widths.
* Styling of story components specific to mobile design mockups is added.

# What side effects does it have?
This PR does not address the height calculation at mobile size, which will be addressed in a future PR.

### CODE REVIEW

File: app/javascript/packs/response-text.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        if (div.children[0].children[0]) {
+          const modifiedDiv = div.children[0].children[0];
+          modifiedDiv.innerText = window.localStorage.getItem(`${div.id}`);
+        }
+      });
+    }
+  }
+
+  function lineClampResponseTexts() {
+    if (document.documentElement.clientWidth > 770) {
```
Comment (mzagaja) [inline app/javascript/packs/response-text.js]: Would this work better with one fo the breakpoints enumerated in our CSS file https://github.com/MAPC/digitalhub/blob/develop/app/assets/stylesheets/global/_vars.scss#L62-L66 or did this have to be 770 for a specific reason?

Comment (smithWEBtek) [reply app/javascript/packs/response-text.js]: That 770 is somewhat arbitrary, I changed it to 670, to correspond to `$break_small: 670px;`

File: app/assets/stylesheets/refinery/stories.scss [hunk 2]
```diff
@@ -147,29 +173,86 @@
       vertical-align: middle;
       width: 24rem;
 
+      &-text {
+        @include media(small) {
+          color: $v3_color_bg-lightest;
+          font-size: $font_size-small;
+          line-height: 1em;
```
[the remark this answers is not in the document]
Comment (smithWEBtek) [reply app/assets/stylesheets/refinery/stories.scss]: Either works in this context, but I will change to rem.

File: app/assets/stylesheets/refinery/stories.scss [hunk 3]
```diff
@@ -125,6 +133,14 @@
     }
 
     &--prompt-small {
+      @include media(small) {
+        font-size: $v3_font_size-event-title-mobile;
+        height: 3.75rem;
+        margin-bottom: 1rem;
```
[the remark this answers is not in the document]
Comment (smithWEBtek) [reply app/assets/stylesheets/refinery/stories.scss]: Changed both small and large to margin 0 at mobile size.

File: app/assets/stylesheets/refinery/stories.scss [hunk 4]
```diff
@@ -139,6 +155,16 @@
...
+      @include media(small) {
+        color: $v3-color_green-dark;
+        font-size: $v3_font_size-event-title-mobile;
+        height: 3.75rem;
+        margin-bottom: 1rem;
+        padding: 1rem;
+        width: 11.9rem;
+      }
+
+      background-image: image-url('middlesex-fells-credit-paul-w.jpg');
```
[the remark this answers is not in the document]
Comment (smithWEBtek) [reply app/assets/stylesheets/refinery/stories.scss]: Changed to full color image, to see contrast of background image.

**Pull request merged**